File: frontend/lib/data/dtos/register_organization_dto.dart
class RegisterOrganizationDto {
  final String organizationName;
  final String? organizationAddress;
  final String adminFirstName;
  final String adminLastName;
  final String adminEmail;
  final String adminPassword;

  RegisterOrganizationDto({
    required this.organizationName,
    this.organizationAddress,
    required this.adminFirstName,
    required this.adminLastName,
    required this.adminEmail,
    required this.adminPassword,
  });

  Map<String, dynamic> toJson() {
    return {
      'organizationName': organizationName,
      'organizationAddress': organizationAddress,
      'adminFirstName': adminFirstName,
      'adminLastName': adminLastName,
      'adminEmail': adminEmail,
      'adminPassword': adminPassword,
    };
  }
}

File: Application/DTOs/RegisterOrganizationDto.cs
using System.ComponentModel.DataAnnotations;

namespace Application.DTOs;

public class RegisterOrganizationDto
{
    [Required]
    [MaxLength(100)]
    public string OrganizationName { get; set; } = string.Empty;

    public string? OrganizationAddress { get; set; }

    [Required]
    [MaxLength(50)]
    public string AdminFirstName { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string AdminLastName { get; set; } = string.Empty;

    [Required]
    [EmailAddress]
    public string AdminEmail { get; set; } = string.Empty;
    
    [Required]
    [MinLength(6)]
    public string AdminPassword { get; set; } = string.Empty;
}
